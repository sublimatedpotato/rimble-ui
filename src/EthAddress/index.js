import React from 'react';
import PropTypes from 'prop-types';
import Text from '../Text';

const addressSummary = (address, firstSegLength = 5, lastSegLength = 4) => {
  if (!address) return '';
  return address
    ? address.slice(0, firstSegLength) +
        '...' +
        address.slice(address.length - lastSegLength)
    : '...';
};

class EthAddress extends React.PureComponent {
  render() {
    const { address, truncate } = this.props;
    const displayAddress = truncate ? addressSummary(address) : address;

    return <Text>{displayAddress}</Text>;
  }
}

EthAddress.propTypes = {
  /**
   * The Ethereum Address
   */
  address: PropTypes.string.isRequired,
  /**
   * Will shorten the address when true
   */
  truncate: PropTypes.bool,
};

EthAddress.defaultProps = {
  truncate: false,
};

EthAddress.displayName = 'EthAddress';

export default EthAddress;
