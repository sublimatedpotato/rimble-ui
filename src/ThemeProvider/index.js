import React from 'react';
import PropTypes from 'prop-types';

import styled, {
  createGlobalStyle,
  ThemeProvider as StyledThemeProvider,
} from 'styled-components';

import { normalize, transparentize } from 'polished';

import theme from '../theme';

export const Base = styled.div`
  & {
    font-size: 1em;
    font-family: ${props => props.theme.fonts.sansSerif};
    color: ${props => props.theme.colors.copyColor};
  }
`;

const GlobalStyle = createGlobalStyle`
  ${normalize()}
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  ::selection {
    background: ${props => transparentize(0.7, props.theme.colors.primary)};
  }
`;

const ThemeProvider = ({ ...props }) => {
  return (
    <StyledThemeProvider theme={theme} {...props}>
      <React.Fragment>
        <GlobalStyle />
        <Base {...props} />
      </React.Fragment>
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {};

export default ThemeProvider;
