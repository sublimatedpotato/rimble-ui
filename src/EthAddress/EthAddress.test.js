import React from 'react';
import { render } from 'enzyme';

import EthAddress from '.';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

const testAddress = '0x99cb784f0429efd72wu39fn4256n8wud4e01c7d2';

describe('Simple Address component sanity', () => {
  it('has name', () => {
    expect(EthAddress.displayName).toBe('EthAddress');
  });

  it('matches default snapshot', () => {
    const component = <EthAddress address={testAddress} />;
    expect(component).toMatchSnapshot();
  });

  it('matches Themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <EthAddress address={testAddress} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
