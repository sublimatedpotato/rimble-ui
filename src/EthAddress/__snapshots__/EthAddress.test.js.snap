// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Address component sanity matches Themed snapshot 1`] = `
.c2 {
  color: #3F3D4B;
}

.c2 {
  box-sizing: border-box;
}

.c1 {
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  text-align: left;
  color: #3F3D4B;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
    color="copyColor"
    font-family="sansSerif"
    font-size="2"
    font-weight="2"
  >
    0x99cb784f0429efd72wu39fn4256n8wud4e01c7d2
  </div>
</div>
`;

exports[`Simple Address component sanity matches default snapshot 1`] = `
<EthAddress
  address="0x99cb784f0429efd72wu39fn4256n8wud4e01c7d2"
  truncate={false}
/>
`;
